DEVELOPMENT.md

```diff
@@ -40,33 +40,47 @@ Automated `unit` tests can be written using [vscode-tmgrammar-test](https://gith
 To start the test suite from the command-line run:
 
 ```bash
-npm test:unit
+npm test
 ```
 
-## Writing Grammar Unit Tests
-
-Unit tests are Terraform files with `vscode-tmgrammar-test` token lines specifying which TextMate grammer should be resolved.
-
-For example:
+## Writing Grammar Snapshot Tests
 
-> Note: This is shortened to demonstrate, actual syntax will vary
+Snapshot tests comprise of two files: example files and their companion `snap` file:
 
+```bash
+tests/snapshot/terraform/blocks.tf
+tests/snapshot/terraform/blocks.tf.snap
 ```
-resource "example" "thing" {
-;         ^^^^^^^ source.terraform string.quoted.double.terraform
-;                   ^^^^^ source.terraform string.quoted.double.terraform
-;                          ^ source.terraform punctuation.section.block.begin.terraform
-}
-; <- source.terraform punctuation.section.block.end.terraform
 
-```
+These files are grouped by HashiCorp product inside folders named for the product they target.
 
-## Writing Grammar Snapshot Tests
+Snapshot test example files are HashiCorp product files without any `vscode-tmgrammar-test` token lines. Each example file is exactly how you would see it used in production. This ensures scope, inheritance, and resolution of tokens happen exactly as they would on a user's machine.
```

Comment:
> Each example file is exactly how you would see it used in production.

I think this is more of an aspiration than reality currently?

We have plenty of tests which contain expressions in the body, which is not valid in any HCL config, e.g. https://github.com/hashicorp/syntax/blob/main/tests/snapshot/hcl/expressions_operators.hcl

Reply:
We can update the existing configs on case-by-case basis, I started with a few files in https://github.com/hashicorp/syntax/pull/44